File: templates/python/quiver_3d.py
"""quiver_3d: 三维矢量场."""
import sys, os; sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
import numpy as np
import matplotlib.pyplot as plt
from theme import apply_theme

def make_figure(title='3D quiver field'):
    apply_theme(fig_size=(6.5, 5))
    x, y, z = np.meshgrid(np.linspace(-2, 2, 6), np.linspace(-2, 2, 6), np.linspace(-2, 2, 6))
    u, v, w = -y, x, 0.3*z
    fig = plt.figure()
    ax = fig.add_subplot(111, projection='3d')
    ax.quiver(x, y, z, u, v, w, length=0.3, normalize=True, color='#0072B2', alpha=0.7)
    ax.set_xlabel('x'); ax.set_ylabel('y'); ax.set_zlabel('z')
    ax.set_title(title)
    fig.tight_layout()
    return fig

if __name__ == '__main__':
    fig = make_figure(); fig.savefig(__file__.replace('.py', '.png'), dpi=150)

File: templates/matlab/quiver_3d.m
function fig = quiver_3d()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    apply_theme();
    [x, y, z] = meshgrid(linspace(-2, 2, 6), linspace(-2, 2, 6), linspace(-2, 2, 6));
    u = -y; v = x; w = 0.3*z;
    fig = figure('Position',[100 100 650 500]);
    quiver3(x, y, z, u, v, w, 1.5, 'Color', palette('cat',1));
    xlabel('x'); ylabel('y'); zlabel('z'); title('3D quiver field');
    view(45, 30); grid on;
end
